### PR TITLE
Add compare with parent action (closes #1283)

### LINKED ISSUE #1283: Feature to compare branches
It would be great to have a feature where you right-click on the branch and have the option to compare it with the parent branch.

Having this feature would allow a pre-code review experience before merging.

Intellij, already in the standard GIT tab, allows for selecting and comparing two branches. Adding things like this to the Machete tab should not be hard (I hope).

### PR DESCRIPTION
**Pull request by mkondratek**

### CODE REVIEW

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    val branchName = getNameOfBranchUnderAction(anActionEvent);
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+    val managedBranch = getManagedBranchByName(anActionEvent, branchName);
+
+    if (gitRepository == null || managedBranch == null || managedBranch.isRoot()) {
+      return;
+    }
+
+    val parent = managedBranch.asNonRoot().getParent();
+    GitBrancher.getInstance(project).compare(parent.getName(), Collections.singletonList(gitRepository));
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java:65]: Hmmm  does it compare selected branch or rather the current working directory? 🤔 given that the other branch is not specified (i.e. it's not `compareAny` that's called)

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    String description = fmt(getNonHtmlString("action.GitMachete.BaseCompareWithParentAction.description.precise"),
+        parent.getName(),
+        managedBranch.getName());
+    presentation.setDescription(description);
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    FileDocumentManager.getInstance().saveAllDocuments();
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java:53]: Also... why this one is needed? 🤔

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java:53]: If current  working directory is compared... then it'd be justified indeed

Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java:53]: I copied it from git4idea - I assume it is needed

Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java:53]: image

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java (hunk 1 above)
Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java:65]: Have you tried it? 

It compares branches based on commits. Working directory changes are ignored. Here is an example:
image

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java:65]: > Working directory changes are ignored. Here is an example:

Whoops indeed 😓 anyway, another issue:

Hmmm I'm checking the current commit... and getting the following unexpected result on each child of `foo-allow-ownership-link` which is the current branch:

image

The only branch for which I'm getting the expected result is `foo-allow-ownership-link`:

image

Sooo... it looks that the selected branch is ignored (?)

Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java:65]: oh, ofc. I must have been drunk or smth ... 🤦 `compareAny` ofc

Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java]: pls try it now

**Review by PawelLipski — APPROVED** [review]
:benissimo:

**Pull request merged**